### ISSUE TITLE
Move caching functionality from NCAR/esds-funnel into xpersist

### ISSUE THREAD

**Issue by andersy005** [NCAR/xpersist#35]
During last week's @NCAR/xdev meeting we agreed that it makes sense to repurpose xpersist by moving the existing caching functionality from [NCAR/esds-funnel](https://github.com/NCAR/esds-funnel) into xpersist. Moving forwards the development of caching-related functionality will happen in `xpersist`. Once the merge is complete, we'll need to re-implement the `@xpersist` decorator 

- [x] https://github.com/NCAR/xpersist/pull/37
- [ ] #61

Ccing @jhamman

**Comment by kmpaul** [comment]
Reimplementing the `@xpersist` decorator can be done later.  Let's move that to a new issue.
